docs/source/routing/performance/caching/response-caching/observability.mdx

```diff
@@ -0,0 +1,269 @@
+
+For example, display a log containing all subgraph response data that's not cached:
+
+```yaml title="router.yaml"
+telemetry:
+  instrumentation:
+    events:
+      subgraph:
+        response:
+          level: info
+          condition:
+            all:
+              - eq: # Only for subgraph posts
+                  - subgraph_name: true
+                  - static: posts
+              - eq: # If there's no cache hit in this subgraph response
+                  - response_cache: hit
+                  - 0
+```
+
+## Cache debugger
+
+The cache debugger in Apollo Sandbox helps you understand cache behavior during development. 
+
+To use it, run the router with the following minimal configuration:
+
+```yaml title="router.yaml"
+supergraph:
+  introspection: true
+  path: /
+  listen: 0.0.0.0:4000
+homepage:
+  enabled: false
+sandbox: # Enabled sandbox
+  enabled: true
+# Enable response caching globally
+preview_response_cache:
+  enabled: true
+  #highlight-start
+  debug: true # Enable debugging data for the cache debugger. Don't enable this in production.
+  #highlight-end
+  invalidation:
+    listen: 0.0.0.0:4000
+    path: /invalidation
+  subgraph:
+    all:
+      enabled: true
+      # Configure Redis for all subgraphs
+      redis:
+        urls: ["redis://localhost:6379"]
+      invalidation:
+        enabled: true
+        shared_key: ${env.INVALIDATION_SHARED_KEY} # Use environment variable INVALIDATION_SHARED_KEY
+```
+
+- Go to your router instance at its root URL to see Apollo Sandbox:
+
+<img
+  className="screenshot"
+  alt="Sandbox in GraphOS Studio"
+  src="../../../../images/response-cache/sandbox.png"
+/>
+
+- In Sandbox, enable the cache debugger. Click the settings button in the top left and scroll to the bottom to enable it:
+
+<img
+  className="screenshot"
+  alt="Sandbox debugger configuration in GraphOS Studio"
+  src="../../../../images/response-cache/debugger-config.png"
+/>
+
+- In the right panel, open the dropdown at the top of the response data panel and select **Cache debugger**:
+
+<img
+  className="screenshot"
+  alt="Sandbox in GraphOS Studio"
+  src="../../../../images/response-cache/sandbox-dropdown.png"
+/>
+
+- A list of cached or potentially cached entries appears. This list helps you understand the cache status of your data:
+  - If the `Created at` column contains data, the value has been stored in the cache
+  - If the `source` column is `products`, the data for this call was fetched from the `products` subgraph, even if it is now cached
+  - If the `Created at` column is empty, the entry hasn't been cached. This might happen for several reasons (see [Troubleshoot](#troubleshoot)). In this example, the `accounts` subgraph entry isn't cached because it contains private, uncacheable data.
```

Comment:
```suggestion
  - If the `Created at` column is empty, the entry hasn't been cached. This might happen for multiple reasons (see [Troubleshoot](#troubleshoot)). In this example, the `accounts` subgraph entry isn't cached because it contains private, uncacheable data.
```